AmazingTwitchBot.Agent/Program.cs

```diff
@@ -11,71 +11,82 @@
 using AmazingTwitchBot.Agent.Rules.ChatCommands;
 using System.Reflection;
 using System.Linq;
+using System.Security.Cryptography.X509Certificates;
 
 namespace AmazingTwitchBot.Agent
 {
-    public class Program
-    {
-        // link to inspiration TwitchBot https://github.com/kasuken/SonequaBot (super straightforward/small example)
+	public class Program
+	{
+		// link to inspiration TwitchBot https://github.com/kasuken/SonequaBot (super straightforward/small example)
 
-        public static void Main(string[] args)
-        {
-            IConfiguration configuration = new ConfigurationBuilder()
-                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
-                .AddUserSecrets<Program>()
-                .Build();
+		public static void Main(string[] args)
+		{
+			IConfiguration configuration = new ConfigurationBuilder()
+				.AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+				.AddUserSecrets<Program>()
+				.Build();
 
 
-            IServiceCollection services = new ServiceCollection();
-            services.Configure<TwitchConfiguration>(configuration.GetSection(nameof(TwitchConfiguration)));
-            services.Configure<ChatConfiguration>(configuration.GetSection(nameof(ChatConfiguration)));
-            services.AddSingleton<TwitchChatBot>();
+			IServiceCollection services = new ServiceCollection();
+			services.Configure<TwitchConfiguration>(configuration.GetSection(nameof(TwitchConfiguration)));
+			services.Configure<ChatConfiguration>(configuration.GetSection(nameof(ChatConfiguration)));
+			services.AddSingleton<TwitchChatBot>();
 
 
-            //var messages = Assembly.GetAssembly(typeof(IChatMessageRule))
-            //    .GetTypes()
-            //    .Where(x => x.Namespace == "AmazingTwitchBot.Agent.Rules.ChatCommands")
-            //    .Where(x => x.IsClass)
-            //    .Select(x => (IChatMessageRule)Activator.CreateInstance(x))
-            //    ;
+			var messages = Assembly.GetAssembly(typeof(IChatMessageRule))
+				.GetTypes()
+				.Where(x => x.Namespace == "AmazingTwitchBot.Agent.Rules.ChatCommands")
+				.Where(x => x.IsClass)
+				.Where(x => !x.IsAbstract)
+				.Select(x => x);
```

Comment:
This line could be removed, because it selects itself :)

Reply:
@P-Storm I'm not sure what you mean here because without the selectors it would throw errors.

Reply:
Ah, it ends with a ;, this should be moved to the row on top. probably you got an line ending error message:

```c#
				.Where(x => x.Namespace == "AmazingTwitchBot.Agent.Rules.ChatCommands")
				.Where(x => x.IsClass)
				.Where(x => !x.IsAbstract);
```

Reply:
oh I get you now yeah my error of couse the where auto selects.